src/js/game/systems/item_acceptor.js

```diff
@@ -75,23 +56,21 @@ export class ItemAcceptorSystem extends GameSystemWithFilter {
             }
 
             const staticComp = entity.components.StaticMapEntity;
-            for (let animIndex = 0; animIndex < acceptorComp.itemConsumptionAnimations.length; ++animIndex) {
-                const { item, slotIndex, animProgress, direction } = acceptorComp.itemConsumptionAnimations[
-                    animIndex
-                ];
+            acceptorComp.inputs.forEach((values, index) => {

```

Comment:
`.forEach` is slow, use a for loop please, same for the other parts

Reply:
leaving this as it'll be changed anyway when javascript maps are no longer used

Reply:
fixed